### PR TITLE
POPO first party plugin comments

### LINKED ISSUE #711: 📋 [TASK] - Investigate impacts of POPO within Plugin System
### Parent Feature

NA

### Task Description

As an AaC developer, I want to investigate impacts of utilizing the new python object within the Plugin system, so that I can have a more reliable plugin capability and can ensure the updated Plugin system can utilize the new python object appropriately.

### Acceptance Criteria

- [x] Review current Plugin system for impacts of changing referenced object(s) (current unstructured dicts) to new python object.
  - [x] active_context
  - [x] gen_plugin
  - [x] help_dump
  - [x] plugin_management
  - [x] primitive_type_check
- [x] Include flags within the code base for impacted areas of Plugin system.
- [x] Document potential necessary changes to Plugin system to assist with developing new Plugin system.

### Unresolved questions

_No response_

### PR DESCRIPTION
**Pull request by crazynewidea**
# Description

Added comments to highlight changes areas for upcoming POPO migration.

# Linked Items:

Closes/Fixes/Resolves #711 

### Added

- _Describe any new features._

### Changed

- _Describe any changes in existing functionality._

### Deprecated

- _Describe any deprecated features._

### Removed

- _Describe any removed features._

### Fixed

- _Describe any bug fixes._

### Security

- _Describe any security-related changes._

# Checklist:

- [ ] I updated project documentation to reflect my changes.
- [ ] My changes generate no new warnings.
- [ ] I updated new and existing unit tests to account for my changes.
- [x] I linked the associated item(s) to be closed.
- [x] I bumped the version.
- [x] I added the labels corresponding to my changes.

### CODE REVIEW

**Comment by lizzcondrey** [comment]
Make sure to bump the version in the root `__init__.py` of `src/aac` since this is merging to main.

**Comment by lizzcondrey** [comment]
You have validators in this branch that don't need to be. Just do a remove and repush.

**Comment by lizzcondrey** [comment]
> Make sure to bump the version in the root `__init__.py` of `src/aac` since this is merging to main.

Team decided to not bump versions for the comment flagging tasks.

File: python/src/aac/__init__.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import os
 
 
-__version__ = "0.3.15"
+__version__ = "0.3.16"
```
Comment (cvchaparro) [inline python/src/aac/__init__.py:16]: Per our discussion, we should revert the version bump.

```suggestion
__version__ = "0.3.15"
```

File: python/src/aac/plugins/first_party/active_context/active_context_impl.py
Comment (cvchaparro) [inline python/src/aac/plugins/first_party/active_context/active_context_impl.py:1]: For this task, there shouldn't have been changes in the `aac.plugins.first_party` package. Only contents in the `aac.plugins.validators` package should have changed.

File: python/src/aac/plugins/validators/required_fields/_validate_required_fields.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 from typing import Any
 
-from aac.lang.constants import DEFINITION_FIELD_FIELDS, DEFINITION_FIELD_NAME, DEFINITION_FIELD_TYPE    # POPO update
-from aac.lang.definitions.definition import Definition  # POPO update
-from aac.lang.definitions.structure import get_substructures_by_type    # POPO update
-from aac.lang.definitions.type import is_array_type # POPO update
+from aac.lang.constants import DEFINITION_FIELD_FIELDS, DEFINITION_FIELD_NAME, DEFINITION_FIELD_TYPE       # POPO Update
+from aac.lang.definitions.definition import Definition     # POPO Update
+from aac.lang.definitions.structure import get_substructures_by_type       # POPO Update
+from aac.lang.definitions.type import is_array_type    # POPO Update
```
Comment (cvchaparro) [inline python/src/aac/plugins/validators/required_fields/_validate_required_fields.py:7]: This doesn't need to be updated based on POPOs.

**Review by cvchaparro — CHANGES_REQUESTED** [review]

**Comment by cvchaparro** [comment]
Oh shoot. I just realized I reviewed the PR with the assumption that it was for #710.

**Comment by lizzcondrey** [comment]
This is just first party-plugins area, so it should be going to `main` not the `integration/validator` branch. So I corrected the target branch.

File: python/src/aac/plugins/first_party/active_context/active_context_impl.py
Comment (crazynewidea) [reply python/src/aac/plugins/first_party/active_context/active_context_impl.py:1]: resolved per your comment below regarding #710

File: python/src/aac/plugins/validators/required_fields/_validate_required_fields.py (hunk 2 above)
Comment (crazynewidea) [reply python/src/aac/plugins/validators/required_fields/_validate_required_fields.py]: removed as requested

**Review by cvchaparro — CHANGES_REQUESTED** [review]
@crazynewidea Per our discussion in stand up yesterday, please remove the changes from `aac.plugins.validators` from this PR.

**Pull request merged**